app/assets/javascripts/3dbio_viewer/src/domain/repositories/PdbInfoRepository.ts

```diff
@@ -3,5 +3,5 @@ import { PdbId } from "../entities/Pdb";
 import { PdbInfo } from "../entities/PdbInfo";
 
 export interface PdbInfoRepository {
-    get(pdbId: PdbId, canTakeAWhile: () => void): FutureData<PdbInfo>;
+    get(pdbId: PdbId, onProcessDelay: (reason: string) => void): FutureData<PdbInfo>;
```

Comment:
My proposal was to add also here an options (containing onProcessDelay). But TBH, that looks also reasonable. So up to you.